plugins/woocommerce-blocks/assets/js/blocks/product-gallery/frontend.tsx

```diff
@@ -30,72 +28,38 @@ const getContext = ( ns?: string ) =>
 type Store = typeof productGallery & StorePart< ProductGallery >;
 const { state, actions } = store< Store >( 'woocommerce/product-gallery' );
 
-/**
- * Product Gallery supports two contexts:
- * - on-page gallery - may display subset of images.
- * - dialog gallery - displays all of the images.
- * Function returns images per current context.
- */
-const getCurrentImages = ( context: ProductGalleryContext ) => {
-	const { isDialogOpen } = context;
-	return context[
-		isDialogOpen ? 'dialogVisibleImagesIds' : 'visibleImagesIds'
-	];
-};
-
-const getImageIndex = ( context: ProductGalleryContext, imageId: string ) => {
-	const imagesIds = getCurrentImages( context );
-	return imagesIds.indexOf( imageId );
+const getImageIndex = (): number => {
+	return state.imageIds.indexOf( state.imageId );
 };
 
 const disableArrows = (
 	context: ProductGalleryContext,
 	newImageNumber: number
 ) => {
-	const imagesIds = getCurrentImages( context );
 	// One-based index so it ranges from 1 to imagesIds.length.
 	context.disableLeft = newImageNumber === 1;
-	context.disableRight = newImageNumber === imagesIds.length;
+	context.disableRight = newImageNumber === context.imageIds.length;
 };
```

Comment:
I don't like that we're mutating `context` here. Can we make this function return new values and assign them correctly. We also wont need to pass `context`, only `newImageNumber` and `numberOfTotalImages`

Reply:
```
type ArrowState = Pick<ProductGalleryContext, 'disableLeft' | 'disableRight'>;

const getArrowState = (imageNumber: number, totalImages: number): ArrowState => ({
    disableLeft: imageNumber === 1,
    disableRight: imageNumber === totalImages
});
```